src/main/webapp/dev/js/main/instructorHelp.js

```diff
@@ -25,7 +34,319 @@ function bindPanelCollapseLinksAndAnchor() {
+            $(button).remove();
+        } else {
+            pageNum = parseInt(pageNum, 10);
+            if (pageNum === curPage) { // disable current page button and highlight it
+                $(button).removeClass('btn-default hidden-xs');
+                $(button).addClass('btn-primary');
+                $(button).prop('disabled', true);
+            } else if (
+                pageNum === 1
+                || pageNum === numPages
+                || pageNum === curPage - 1
+                || pageNum === curPage + 1
+            ) {
+                $(button).removeClass('hidden-xs');
+                $(button).prop('disabled', false);
+
+                // add ellipses for small screen instead of showing all buttons
+                if ((pageNum === curPage - 1) && (curPage - 2 > 1)) {
+                    $(button).before('<button type="button" class="btn visible-xs" disabled>...</button>');
+                }
+                if ((pageNum === curPage + 1) && (curPage + 2 < numPages)) {
+                    $(button).after('<button type="button" class="btn visible-xs" disabled>...</button>');
+                }
+            } else { // enable all other buttons
+                $(button).addClass('hidden-xs');
+                $(button).prop('disabled', false);
+            }
+        }
+    });
+
+    // disable prev and next buttons for first and last page respectively
+    if (curPage === 1) {
+        $('#prevPage').prop('disabled', true);
+    }
+    if (curPage === numPages) {
+        $('#nextPage').prop('disabled', true);
+    }
+}
+
+/**
+ * Resets the page to default version and clears the search box.
+ */
+function displayDefaultAndClearSearchBox() {
+    $('#searchQuery').val('');
+    $('#searchMetaData').text('');
+    $('#searchResults').empty().hide();
+    $('#pagingControls').empty();
+    $('#pagingControls').removeClass('padding-15px margin-bottom-35px');
+    $('#pagingDivider').hide();
+    $('#allQuestions').show();
+    $('#topics').show();
+}
+
+/**
+ * Searches for the query specified in the search box
+ */
+function searchQuestions() {
+    const query = $('#searchQuery').val();
+
+    $('#searchResults').empty();
+    $('#pagingControls').empty();
+    $('#pagingControls').removeClass('padding-15px margin-bottom-35px');
+    if (query === '') {
+        displayDefaultAndClearSearchBox();
+        return;
+    }
+    // different fields are given a boost value to specify their relative importance
+    // for details refer to elasticlunr documentation
+    const results = index.search(query, {
+        fields: {
+            title: { boost: 4 },
+            tags: { boost: 2 },
+            body: { boost: 0.5 },
+        },
+        bool: 'AND',
+    });
+    // add relevant panels to search results
+    $.each(results, (idx, result) => {
+        const { htmlId } = questionMap[result.ref];
+        const questionPanel = $(`#${htmlId}`)[0];
```

Comment:
actually i just tested it and i do need it since i need to obtain the outerhtml and for that I need the native DOM element. Fastest way of getting it is using indexing as shown below:
https://learn.jquery.com/using-jquery-core/faq/how-do-i-pull-a-native-dom-element-from-a-jquery-object/
If you have a better suggestion please share

Reply:
Can use jQuery to do the manipulation rather than the native API? Though I have no objection to add [0].